src/main/java/com/store/litflix/mapper/BookMapper.java

```diff
@@ -1,18 +1,29 @@
 package com.store.litflix.mapper;
 
 import com.store.litflix.config.MapperConfig;
-import com.store.litflix.dto.book.BookDto;
 import com.store.litflix.dto.book.BookDtoWithoutCategoryIds;
+import com.store.litflix.dto.book.BookResponseDto;
 import com.store.litflix.dto.book.CreateBookRequestDto;
 import com.store.litflix.dto.book.UpdateBookRequestDto;
 import com.store.litflix.model.Book;
+import com.store.litflix.model.Category;
+import java.util.List;
+import java.util.Set;
 import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
 import org.mapstruct.MappingTarget;
+import org.mapstruct.Named;
 
 @Mapper(config = MapperConfig.class)
 public interface BookMapper {
+    @Mapping(target = "categoryIds", source = "categories", qualifiedByName = "mapCategoryIds")
+    BookResponseDto toDto(Book book);
 
-    BookDto toDto(Book book);
+    @Named("mapCategoryIds")
+    static List<Long> mapCategoryIds(Set<Category> categories) {
+        return categories == null ? null :
+                categories.stream().map(com.store.litflix.model.Category::getId).toList();
```

Comment:
use import for Category
```suggestion
                categories.stream()
                .map(com.store.litflix.model.Category::getId)
                .toList();
```